### PR TITLE
Make aircraft takeoff after creation/resupply optional

### PR DESCRIPTION
**Pull request by reaperrr**
Modders shouldn't be forced to hack around in Mods.Common just to replicate the original games' behaviour.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs [hunk 1]
```diff
@@ -72,6 +77,10 @@ public class AircraftInfo : ITraitInfo, IPositionableInfo, IFacingInfo, IMoveInf
 		[Desc("Does this actor cancel its previous activity after resupplying?")]
 		public readonly bool AbortOnResupply = true;
 
+		[Desc("Does this actor automatically take off after being built or after resupplying?")]
+		public readonly bool TakeOffAutomatically = false;
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Um, this should be `true` to keep the current behavior, I guess?

**Review by abcdefg30 — DISMISSED** [review]
Even with `TakeOffAutomatically: true` yaks don't take off after resupplying(?).

File: OpenRA.Mods.Common/Traits/Air/Aircraft.cs (hunk 1 above)
Comment (reaperrr) [inline OpenRA.Mods.Common/Traits/Air/Aircraft.cs]: Sorry, seems the test setting sneaked in, should be true, of course.

**Comment by reaperrr** [comment]
Updated.

**Comment by F5544** [comment]
great work, how can i make this version into a playtest openra.common.dll version, instead of a developer test openra.common.dll?

**Comment by F5544** [comment]
i really like this option and want to add it in my latest playtest. but bleed compile to playtest gives me errors. how can i replace the playtest code with this one?

**Comment by abcdefg30** [comment]
Sorry, you can't integrate this change into the playtest without breaking multiplayer ability or creating your own mod. Otherwise you need to wait until this is released, which could still take a while, as it still needs to be reviewed and merged (and it is not merged into the "stable" `prep-1707` branch, which means this will only come the next playtest *after* the next release). :/

**Review by abcdefg30 — DISMISSED** [review]
This still doesn't work properly with aircraft that use `TakeOffAutomatically: true` (https://github.com/OpenRA/OpenRA/pull/13638#pullrequestreview-50191103). They do take off after being built, but not after being resupplied.

**Comment by reaperrr** [comment]
I split the style/cleanup commits to #13848, as this PR might need a bit more thinking to handle the production/resupply separation properly.

**Comment by F5544** [comment]
will this implemented in the new release?

**Comment by reaperrr** [comment]
> will this implemented in the new release?

Sorry, apart from bugfixes the branch for the upcoming release is in a feature-freeze, so this will not arrive until next release after that.

**Comment by reaperrr** [comment]
Rebased and updated.

**Comment by reaperrr** [comment]
Superceded by #13865.

**Pull request closed without merging**